### ISSUE TITLE
Split button design in Calypso

### ISSUE THREAD

**Issue by eeeeevon13** [Automattic/wp-calypso#43115]
#### Is your feature request related to a problem? Please describe.
While diagnosing a Publicize issue here (https://github.com/Automattic/jetpack/issues/13570), I noticed that our split button does not have a divider line. This makes it look less like a button with two similar actions, and more like a button with an icon that just happens to be in the right. 

#### Describe the solution you'd like
Add a 1px white line (or space) between the left and right buttons in the split.

#### Describe alternatives you've considered
I looked at best practice for split buttons and pulled some good real world examples. The line is really the way to go. IMO.  I also did a version w/a dark line instead but the white one feels crisper. 

**BEFORE**
![before](https://user-images.githubusercontent.com/49159284/84183640-d58bc000-aa51-11ea-8e2c-65cc4349d6c3.jpg)

**AFTER:**
![after](https://user-images.githubusercontent.com/49159284/84183641-d6245680-aa51-11ea-9b34-9d1195843c72.jpg)

**Comment by folletto** [comment]
Yes. There seems to be a bug when the buttons were restyled and the separator was lost.
As you noted, the same issue happens in the devdocs, so it seems a problem at component level: 
https://wpcalypso.wordpress.com/devdocs/design/button-group

<img width="310" alt="Screenshot 2020-06-09 at 19 16 41" src="https://user-images.githubusercontent.com/4389/84184750-d4748600-aa85-11ea-821c-98015c9e179c.png">

**Comment by Aurorum** [comment]
Just noting the previous design here. I suspect the reason a darker colour was used had to do with ensuring it looked good for all colour schemes too.

<img width="288" alt="Screenshot 2020-06-09 at 20 54 00" src="https://user-images.githubusercontent.com/43215253/84193456-62a33900-aa93-11ea-94c5-5fe17bc015d1.png">

Not sure if it's ideal with the new design, but I suspect it could be the same here too. 

<img width="275" alt="Screenshot 2020-06-09 at 20 54 32" src="https://user-images.githubusercontent.com/43215253/84193516-7cdd1700-aa93-11ea-8817-5aa80be16cad.png">
